program/src/instructions/legacy/take_bid.rs

```diff
@@ -62,11 +65,21 @@ pub struct TakeBidLegacy<'info> {
     pub seller_ta: Box<InterfaceAccount<'info, TokenAccount>>,
 
     /// CHECK: whitelist, token::mint in seller_token, associated_token::mint in owner_ata_acc
+    #[account(
+        mint::token_program = token_program,
+    )]
     pub mint: Box<InterfaceAccount<'info, Mint>>,
 
-    //can't deserialize directly coz Anchor traits not implemented
-    /// CHECK: assert_decode_metadata check seeds
-    #[account(mut)]
+    /// CHECK: ownership, structure and mint are checked in assert_decode_metadata, seeds checked here.
+    #[account(mut,
+        seeds = [
+            Metadata::PREFIX,
+            mpl_token_metadata::ID.as_ref(),
+            mint.key().as_ref(),
+        ],
+        bump,
+        seeds::program = mpl_token_metadata::ID,
+    )]
     pub metadata: UncheckedAccount<'info>,
```

Comment:
Didn't we remove this before b/c of stack issues?

Reply:
Don't see any comments about that in the original code. These seeds are checked in Token Metadata but added them here for consistency.